zql/docs/processors/README.md

```diff
@@ -171,9 +171,9 @@ ID.ORIG_H     ID.ORIG_P ID.RESP_H       ID.RESP_P ORIG_BYTES RESP_BYTES TOTAL_BY
 |                           |                                                                           |
 | ------------------------- | ------------------------------------------------------------------------- |
 | **Description**           | Sort events based on the order of values in the specified named field(s). | 
-| **Syntax**                | `sort [-r] [-limit N] [-nulls first\|last] [field-list]`                   |
+| **Syntax**                | `sort [-r] [-limit MB] [-nulls first\|last] [field-list]`                 |
```

Comment:
I think we should remove the limit parameter entirely. In this form, as a user supplied size of memory, it's both dangerous, because a nefarious user could use a very large size to OOM the service, and it's hard to use, since a user won't know how much is too little or too much. 

I think a configurable cutoff for when to shift from memory-only to disk usage does make sense as a config parameter for zqd/zq, though, and that can be memory based, since the operator would know more about the machine where zqd/zq is running.

Reply:
👍

Reply:
@alfred-landrum: I replaced the limit parameter with, proc.SortMemMaxBytes, a global variable set by zq in response to `-sortmem` and set by zqd in response to `sortMemMaxBytes` in the config file.

@alfred-landrum and @mattnibs: I'm a little uncomfortable adding a global variable. The only reasonable alternative I see is passing a configuration struct or a configuration function to proc.Compile and proc.CompileProc, but that's a little more complicated than the global variable. Let me know if you think it's worth doing.